src/java.base/unix/native/libnio/fs/UnixNativeDispatcher.c

```diff
@@ -545,12 +705,26 @@ Java_sun_nio_fs_UnixNativeDispatcher_lstat0(JNIEnv* env, jclass this,
     int err;
     struct stat64 buf;
     const char* path = (const char*)jlong_to_ptr(pathAddress);
-
+#if defined(__linux__)
+    struct my_statx statx_buf;
+    int flags = AT_STATX_SYNC_AS_STAT;
+    unsigned int mask = STATX_ALL;
+
+    if (my_statx_func != NULL) {
+        // Prefer statx over stat64 on Linux if it's available
+        RESTARTABLE(statx_wrapper(AT_FDCWD, path, flags, mask, &statx_buf, NO_FOLLOW_SYMLINK), err);
+        if (err == 0) {
+            copy_statx_attributes(env, &statx_buf, attrs);
+        } else {
+            throwUnixException(env, errno);
+        }
+    }
+#endif
```

Comment:
Are you missing a return here or did you really intent to call both  `statx` and `lstat64` ?

Reply:
Good catch! The original code had this in `if () ... else ...` blocks. Will fix. It needs to get fixed elsewhere too.